community/blog/2023-02 Integrating Arcus Security in a non-invasive manner/README.md

```diff
@@ -0,0 +1,134 @@
+    .ConfigureAppConfiguration(builder =>
+    {
+        builder.AddEnvironmentVariables();
+    })
+    .ConfigureServices((context, services) =>
+    {
+        services.AddAzureClients(clients =>
+        {
+            string serviceUrl = context.Configuration["AZURE_BLOB_URL"];
+            string accountName = context.Configuration["AZURE_BLOB_ACCOUNTNAME"];
+            string accountKey = context.Configuration["AZURE_BLOB_ACCOUNTKEY"];
+
+            clients.AddClient<BlobServiceClient, BlobClientOptions>((options, serviceProvider) =>
+            {
+                return new BlobServiceClient(new Uri(serviceUrl), new StorageSharedKeyCredential(accountName, accountKey));
+            });
+        });
+    })
+    .Build();
+
+host.Run();
+```
+
+This Azure Blob storage client is injected in our Azure Function HTTP trigger:
+
+```csharp
+using System.Net;
+using Azure.Storage.Blobs;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Logging;
+
+public class StoreFunction
+{
+    private readonly BlobServiceClient _blobClient;
+    private readonly ILogger _logger;
+
+    public StoreFunction(
+        IAzureClientFactory<BlobServiceClient> clientFactory,
+        ILoggerFactory loggerFactory)
+    {
+        _blobClient = clientFactory.CreateClient("Default");
+        _logger = loggerFactory.CreateLogger<StoreFunction>();
+    }
+
+    [Function("store")]
+    public HttpResponseData Run(
+        [HttpTrigger(AuthorizationLevel.Function, "post")] HttpRequestData request)
+    {
+        _logger.LogInformation("C# HTTP trigger function processed a request");
+
+        // Save something to Azure Blob storage with client...
+        
+        return request.CreateResponse(HttpStatusCode.Accepted);
+    }
+}
+```
+
+## Separating sensitive secrets from application configuration
+The trick is to find the secrets in your application configuration and treat them differently. [We have a whole section](https://security.arcus-azure.net/features/secret-store) in our feature documentation that describes why Arcus is recommending this. Why it is safer, cleaner, and more maintainable. It is important to realize that Arcus does not act as a 'nice to have' at this point, but brings real value to the table which is both critical and necessary. ([Learn more about the dangers of solely using application configuration](https://www.codit.eu/blog/introducing-secret-store-net-core/)).
+
+The Azure Blob storage account key is the main secret in this example. It is important to remember that in some scenarios both configuration and secrets would be in the same place, but that does not mean that they should reside in the same place in your application code. Treating secrets as secrets and not public configuration data is a big selling point here. It means that we mentally work more carefully where we retrieve secret information and where we retrieve common data.
```

Comment:
Normally, config & secrets should be separated instead of keeping them in the same place

Reply:
Yes, I agree. 👍
It is only based on things I've seen in Invictus and wanted to show (via blog) what the first step could be.

Reply:
Removed the 'It is important to remember...' to 'Ideally, we store in separate places...'.